homeassistant/data_entry_flow.py

```diff
@@ -84,7 +84,8 @@ class FlowResultType(StrEnum):
 }
 
 
-_FlowResultT = TypeVar("_FlowResultT", bound="FlowResult")
+_FlowResultT = TypeVar("_FlowResultT", bound="BaseFlowResult")
+_HandlerT = TypeVar("_HandlerT")
```

Comment:
```suggestion
_HandlerT = TypeVar("_HandlerT", default=str)
```
The `handler` type is almost always `str`. With the `mypy-dev` PR merged, using `default=str` is now possible. Just remember to import `TypeVar` from `typing_extensions` instead.

If you like, after the change you could go around and remove the unnecessary `str` types.
```diff
-BaseFlowHandler[ConfigFlowResult, str]
+BaseFlowHandler[ConfigFlowResult]
```

Reply:
Yes, indeed 👍 

Likewise, I think the `_FlowResultT` can default to `FlowHandler` and we can remove the BaseXXX types?

Reply:
> Likewise, I think the `_FlowResultT` can default to `FlowHandler`

The `default` needs to be a subtype of the `bound` type. So this must be `BaseFlowResult`, `FlowResult`, or similar. Not sure it does make sense though as it's frequently subclassed -> `ConfigFlowResult`, `AuthFlowResult`.

> [...] and we can remove the BaseXXX types?

Not sure what you mean by this.